bundles/org.openhab.core/src/test/java/org/eclipse/smarthome/core/library/types/DateTimeTypeTest.java

```diff
@@ -232,79 +314,105 @@ public void parsingTest() {
+                    parameterSet.inputTimeMap.get("month") + 1, parameterSet.inputTimeMap.get("date"),
+                    parameterSet.inputTimeMap.get("hourOfDay"), parameterSet.inputTimeMap.get("minute"),
+                    parameterSet.inputTimeMap.get("second"), durationInNano);
+            ZonedDateTime zonedDate = ZonedDateTime.of(dateTime, parameterSet.inputTimeZone.toZoneId());
+            dt1 = new DateTimeType(zonedDate);
+            dt3 = new DateTimeType(
+                    zonedDate.format((DateTimeFormatter.ofPattern(DateTimeType.DATE_PATTERN_WITH_TZ_AND_MS))));
+            zonedDate = ZonedDateTime.of(dateTime, parameterSet.inputTimeZone.toZoneId()).toInstant()
+                    .atZone(parameterSet.defaultTimeZone.toZoneId());
+            dt2 = new DateTimeType(zonedDate);
+        } else if (parameterSet.inputTimeString != null) {
+            dt1 = new DateTimeType(parameterSet.inputTimeString);
+            dt2 = new DateTimeType(dt1.getZonedDateTime().withZoneSameInstant(TimeZone.getDefault().toZoneId()));
+            dt3 = new DateTimeType(dt1.getZonedDateTime());
         } else {
-            inputTimeString = parameterSet.inputTimeString;
-        }
-        DateTimeType dt = DateTimeType.valueOf(inputTimeString);
-        if (parameterSet.inputTimeZone == null) {
-            dt = new DateTimeType(dt.getZonedDateTime().withZoneSameInstant(TimeZone.getDefault().toZoneId()));
+            return;
         }
         // Test
-        assertEquals(parameterSet.expectedResult, dt.toString());
+        assertEquals(dt1.toFullString(), dt1.toString());
+        assertEquals(dt2.toFullString(), dt2.toString());
+        assertEquals(parameterSet.expectedResult, dt1.toString());
+        assertEquals(parameterSet.expectedResultLocalTZ, dt2.toString());
+        assertEquals(parameterSet.expectedResult, dt3.toString());
+        assertEquals(dt1, dt3);
     }
 
     @Test
-    public void createZonedDate() {
-        String inputTimeString;
+    public void formattingTest() {
+        DateTimeType dt = createDateTimeType();
+        if (parameterSet.locale != null) {
+            assertEquals(parameterSet.expectedFormattedResult, dt.format(parameterSet.locale, parameterSet.pattern));
+        } else {
+            assertEquals(parameterSet.expectedFormattedResult, dt.format(parameterSet.pattern));
+        }
+    }
 
-        // set default time zone
-        TimeZone.setDefault(parameterSet.defaultTimeZone);
+    @Test
+    public void changingZoneTest() {
+        DateTimeType dt = createDateTimeType();
+        DateTimeType dt2 = dt.toLocaleZone();
+        assertEquals(parameterSet.expectedResultLocalTZ, dt2.toFullString());
+        dt2 = dt.toZone(parameterSet.defaultTimeZone.toZoneId());
+        assertEquals(parameterSet.expectedResultLocalTZ, dt2.toFullString());
+        boolean thrown = false;
+        try {
+            dt2 = dt.toZone("XXX");
+        } catch (DateTimeException e) {
```

Comment:
This should be an own test which expects the thrown exception:

```
    @Test(expected = DateTimeException.class)
    public void changingZoneThrowsExceptionTest() {
        // TODO
    }
```

Reply:
Done